packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -0,0 +1,259 @@
+      props.text = children?.trim().toUpperCase();
+      break;
+    default:
+  }
+
+  if (isDisabled) {
+    const disabledColor = getIn(theme.colors, 'action.background.primary.disabled');
+    const disabledBorderColor = getIn(theme.colors, 'action.border.primary.disabled');
+    props.iconColor = 'action.icon.primary.disabled';
+    props.textColor = 'action.text.primary.disabled';
+    props.color = disabledColor;
+    props.borderColor = disabledBorderColor;
+    props.hoverColor = disabledColor;
+    props.hoverBorderColor = disabledBorderColor;
+    props.activeColor = disabledColor;
+    props.activeBorderColor = disabledBorderColor;
+    props.focusColor = disabledColor;
+    props.focusBorderColor = disabledBorderColor;
+  }
+
+  return props;
+};
+
+const BaseButton = ({
+  variant = 'primary',
+  intent,
+  contrast = 'low',
+  size = 'medium',
+  icon: Icon,
+  iconPosition = 'left',
+  isDisabled = false,
+  isFullWidth = false,
+  onClick,
+  type = 'button',
+  children,
+}: BaseButtonProps): ReactElement => {
+  const { theme, platform } = useTheme();
+  const buttonTypographyTokens = typography[platform];
+  if (!Icon && !children?.trim()) {
+    throw new Error(`[Blade: BaseButton]: Cannot render a BaseButton without an icon or text`);
+  }
+  const {
+    activeBorderColor,
+    activeColor,
+    borderColor,
+    color,
+    minHeight,
+    spacing,
+    focusBorderColor,
+    focusColor,
+    focusRingColor,
+    fontSize,
+    hoverBorderColor,
+    hoverColor,
+    iconColor,
+    iconSize,
+    iconSpacing,
+    lineHeight,
+    text,
+    textColor,
+    borderWidth,
+    borderRadius,
+  } = getProps({
+    buttonTypographyTokens,
+    children,
+    isDisabled,
+    size,
+    theme,
+  });
+  console.log('unused props', {
+    variant,
+    intent,
+    contrast,
+  });
+  return (
+    <StyledBaseButton
+      activeBorderColor={activeBorderColor}
+      activeColor={activeColor}
+      borderColor={borderColor}
+      minHeight={minHeight}
+      spacing={spacing}
+      color={color}
+      disabled={isDisabled}
+      focusBorderColor={focusBorderColor}
+      focusColor={focusColor}
+      focusRingColor={focusRingColor}
+      hoverBorderColor={hoverBorderColor}
+      hoverColor={hoverColor}
+      isFullWidth={isFullWidth}
+      onClick={onClick}
+      type={type}
+      borderWidth={borderWidth}
+      borderRadius={borderRadius}
+    >
+      {Icon && iconPosition == 'left' ? <Icon size={iconSize} color={iconColor} /> : null}
+      {text && (
+        <StyledBaseText
+          lineHeight={lineHeight}
+          fontSize={fontSize}
+          fontWeight="bold"
+          textAlign="center"
+          color={textColor}
+          iconPosition={iconPosition}
+          hasIcon={!!Icon}
+          iconSpacing={iconSpacing}
+        >
+          {text}
+        </StyledBaseText>
+      )}
+      {Icon && iconPosition == 'right' ? <Icon size={iconSize} color={iconColor} /> : null}
```

Comment:
the icon right is not working on the storybook, can you please check? Also not sure what all props are functional on the storybook, a couple of them don't work. 


<img width="586" alt="Screenshot 2022-06-14 at 12 33 31 PM" src="https://user-images.githubusercontent.com/11384858/173514423-2ae7b2a9-e6a0-4d01-b3d3-1eda823706f3.png">

Reply:
Working for me 🤔 

Except `contrast` & `intent`, all other props should work - as they are on localhost for me. CI Is failing, not able to check if its working there (I'll debug the failing CI)
<img width="902" alt="Screenshot 2022-06-15 at 10 40 22 AM" src="https://user-images.githubusercontent.com/24487274/173741217-2aef4e45-b977-42dc-9741-9bd794d811b3.png">

Reply:
CI is working but just realised that you cant see it in CI since its a prod build and this is an internal story. Can you check again on your local?